lib/ansible/modules/cloud/packet/packet_device.py

```diff
@@ -562,10 +614,10 @@ def main():
     state = module.params.get('state')
 
     try:
-        module.exit_json(**act_on_devices(state, module, packet_conn))
+        module.exit_json(**act_on_devices(module, packet_conn, state))
     except Exception as e:
-        module.fail_json(msg='failed to set machine state %s, error: %s' % (state,str(e)))
-
+        module.fail_json(msg='failed to set machine state %s, error: %s'
+                         % (state, e))
```

Comment:
When catching a generic exception, not knowing why it failed; you should include the stacktrace.  It's also safer to manually transform the exception into a strig usig to_native.  So this code will look like this:
``` python
import traceback

from ansible.module_utils._text import to_native
[...]

module.fail_json(msg='failed to set machine state %s, error: %s'
                           % (state, to_native(e)), exception=traceback.format_exc())